raiden/raiden_service.py

```diff
@@ -1117,6 +1127,7 @@ def start_mediated_transfer_with_secret(
         identifier: PaymentID,
         secret: Secret,
         secrethash: SecretHash = None,
+        locktimeout: BlockTimeout = ABSENT_BLOCKTIMEOUT,
```

Comment:
IMO we should maintain the type `Optional[BlockTimeout]` here. And if the value `is None` then `get_safe_initial_expiration` is called, otherwise we use the user's provided value.

Reply:
@hackaugusto 

Initially I just used `None` for missing `locktimeout`. This works across the code but fail with restoring `TransferDescriptionWithSecretState` from the database. Looks like the `dataclass` is not ready to decode None value.

For that reason I had to use a value when locktimeout is not given. 

I will create a new type for locktimeout (not to be confused with blocktimeout) and create an ABSENT_LOCKTIMEOUT with a value of 0.

Ok?

Reply:
> Initially I just used None for missing locktimeout. This works across the code but fail with restoring TransferDescriptionWithSecretState from the database. Looks like the dataclass is not ready to decode None value.

I can help you with the restoration if i know exactly what the problem is.